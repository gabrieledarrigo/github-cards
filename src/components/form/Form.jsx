import React from 'react';
import PropTypes from 'prop-types';

export default function Form({ onSubmit, children, ...rest }) {
  return (
    <form
      {...rest}
      onSubmit={(e) => {
        e.preventDefault();
        onSubmit();
      }}
    >
      {children}
    </form>
  );
}

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.element.isRequired,
};
