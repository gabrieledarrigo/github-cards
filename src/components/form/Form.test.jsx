import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Form onSubmit={() => {}}>
          <div />
        </Form>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should invoke onSubmit function when a user submits the form', () => {
    const onSubmit = jest.fn();
    const preventDefault = jest.fn();
    const wrapper = shallow(
      <Form onSubmit={onSubmit}>
        <div>form</div>
      </Form>,
    );

    wrapper.find('form').simulate('submit', {
      preventDefault,
    });

    expect(onSubmit).toHaveBeenCalled();
  });
});
