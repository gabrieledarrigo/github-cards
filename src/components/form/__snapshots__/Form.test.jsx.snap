// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div />
</form>
`;
